src/rx_signal_processing.py

```diff
@@ -182,76 +214,86 @@ def sequence_worker(options, ringbuffer):
 
         # Tell brian DSP is about to begin
         mark_timer = time.perf_counter()
-        reply_packet = {"sequence_num": sequence_num}
+        reply_packet = {"sequence_num": rx_params.sequence_num}
         msg = pickle.dumps(reply_packet, protocol=pickle.HIGHEST_PROTOCOL)
         so.recv_bytes(dspbegin_to_brian, options.brian_to_dspbegin_identity, log)
         so.send_bytes(dspbegin_to_brian, options.brian_to_dspbegin_identity, msg)
         log_dict["dsp_begin_msg_time"] = (time.perf_counter() - mark_timer) * 1e3
 
-        if cfs_scan_flag:
+        if rx_params.cfs_scan_flag:
             # CFS analysis
             mark_timer = time.perf_counter()
             cfs_processor = DSP(
-                rx_rate,
-                filter_taps,
-                mixing_freqs,
-                downsample_rates,
+                rx_params.rx_rate,
+                rx_params.filter_taps,
+                rx_params.mixing_freqs,
+                rx_params.downsample_rates,
                 use_shared_mem=False,
             )
             cfs_processor.apply_filters(sequence_samples)
             cfs_processor.move_filter_results()
-            cfs_data, cfs_freq = cfs_processor.cfs_freq_analysis(slice_details)
+            cfs_data, cfs_freq = cfs_processor.cfs_freq_analysis(
+                rx_params.slice_details[0]
+            )
             log_dict["cfs_dsp_time"] = (time.perf_counter() - mark_timer) * 1e3
 
         else:
             # Process main samples
             mark_timer = time.perf_counter()
             main_sequence_samples = sequence_samples[: len(options.rx_main_antennas), :]
             main_sequence_samples_shape = main_sequence_samples.shape
-            main_processor = DSP(rx_rate, filter_taps, mixing_freqs, downsample_rates)
+            main_processor = DSP(
+                rx_params.rx_rate,
+                rx_params.filter_taps,
+                rx_params.mixing_freqs,
+                rx_params.downsample_rates,
+            )
             main_processor.apply_filters(main_sequence_samples)
             main_processor.move_filter_results()
-            main_processor.beamform(main_beam_angles)
+            main_processor.beamform(rx_params.main_beam_angles)
             main_corrs = DSP.correlations_from_samples(
                 main_processor.beamformed_samples,
                 main_processor.beamformed_samples,
-                output_sample_rate,
-                slice_details,
+                rx_params.output_sample_rate,
+                rx_params.slice_details,
             )
             log_dict["main_dsp_time"] = (time.perf_counter() - mark_timer) * 1e3
 
             # Process intf samples if intf exists
             mark_timer = time.perf_counter()
             intf_sequence_samples_shape = None
-            log_dict["intf antennas"] = intf_antennas
-            if len(intf_antennas) > 0:
+            log.info("intf antennas", antennas=rx_params.intf_antennas)
```

Comment:
Wait no change this back, it should be included in log_dict so it only logs at a verbose level, not info